src/PathFactory.js

```diff
@@ -50,10 +50,19 @@ export default class PathFactory {
     if (!data)
       [data, settings] = [settings, null];
 
+    // Set data as subject if input as string
+    if (typeof data === 'string')
+      data = { subject: data };
+
     // Apply defaults on settings and data
-    return this._pathProxy.createPath(
-      Object.assign(Object.create(null), this._settings, settings),
-      Object.assign(Object.create(null), this._data, data));
+    const _settings = { ...this._settings, ...settings };
+    const _data = { ...this._data, ...data };
+
+    // Resolve string subjects to namedNodes
+    if (typeof _data.subject === 'string')
+      _data.subject = new JSONLDResolver(_settings.context).lookupProperty(_data.subject);
```

Comment:
I'm a tiny bit surprised that `JSONLDResolver` is hardcoded here… is there a better way?

Reply:
i.e. reuse the one from line 29?